include/deal.II/base/utilities.h

```diff
@@ -1321,10 +1377,61 @@ namespace Utilities
       (void)cend;
 
       // Then copy the elements:
-      if (object.size() > 0)
-        memcpy(object.data(),
-               &*cbegin + sizeof(vector_size),
-               vector_size * sizeof(T));
+      object.clear();
+      if (vector_size > 0)
+        object.insert(object.end(),
+                      reinterpret_cast<const T *>(&*cbegin +
+                                                  sizeof(vector_size)),
+                      reinterpret_cast<const T *>(&*cend));
```

Comment:
Out of curiosity, why not use `memcpy`?

Reply:
The motivation is that if you first `resize()` an object to its size and then call `memcpy`, you need to touch the data twice, once to place whatever default value the data container is going to have, and then to copy the actual content in place. Running `insert` avoids one of those passes. Here it would probably not have mattered a lot indeed because most of the vectors we have are small, as the sit inside another vector, but overall the memory transfer can only get better. Furthermore, the new way should never make things worse than `memcpy`: Most STL implementations I know of internally call `memcpy` to fill the data of trivial types, so `memcpy` will just be called somewhere else. There might be a small overhead of a few instructions for one versus the other way, but I would not see it significant to bother about that constant-time cost.

Reply:
Ah, nice, I didn't think about the touch-twice issue!